### PR TITLE
Add documentation about B3 header use with go.

### PR DESCRIPTION
**Pull request by knusbaum**
### What does this PR do?
This PR adds documentation describing the use of the B3 headers with Go.

### Motivation
This is part of an effort to document B3 header use for all languages that implement it.

### Preview link
https://docs-staging.datadoghq.com/knusbaum/go-b3-headers/tracing/setup/go

This documentation is lifted nearly word-for-word from [the Java section](https://docs.datadoghq.com/tracing/setup/java/#b3-headers-extraction-and-injection), but I've verified that everything here is correct for the go tracer.

### CODE REVIEW

File: content/en/tracing/setup/go.md [hunk 1]
```diff
@@ -139,6 +139,35 @@ func main() {
...
+
+Distributed headers injection and extraction is controlled by
+configuring injection/extraction styles. Currently two styles are
+supported:
+
+* Datadog: `Datadog`
+* B3: `B3`
+
+Injection styles can be configured using the environment Variable
+`DD_PROPAGATION_STYLE_INJECTION=Datadog,B3`
```
Comment (gbbr) [inline content/en/tracing/setup/go.md]: It's called 
```suggestion
`DD_PROPAGATION_STYLE_INJECT=Datadog,B3`
```

File: content/en/tracing/setup/go.md [hunk 2]
```diff
@@ -139,6 +139,35 @@ func main() {
...
+Datadog APM tracer supports [B3 headers extraction][58] and injection for distributed tracing.
+
+Distributed headers injection and extraction is controlled by
+configuring injection/extraction styles. Currently two styles are
+supported:
+
+* Datadog: `Datadog`
+* B3: `B3`
+
+Injection styles can be configured using the environment Variable
```
Comment (gbbr) [inline content/en/tracing/setup/go.md]:
```suggestion
Injection styles can be configured using the environment variable
```

File: content/en/tracing/setup/go.md [hunk 3]
```diff
@@ -139,6 +139,35 @@ func main() {
...
+The value of the property or environment variable is a comma separated list of
+header styles that are enabled for injection. By default only Datadog injection
+style is enabled.
+
+Extraction styles can be configured using the environment Variable
+`DD_PROPAGATION_STYLE_EXTRACT=Datadog,B3`
+
+The value of the property or environment variable is a comma separated list of
+header styles that are enabled for extraction. By default onlyDatadog extraction
+style is enabled.
```
Comment (gbbr) [inline content/en/tracing/setup/go.md]: Can't we merge these so we don't repeat the same paragraph?

Comment (knusbaum) [reply content/en/tracing/setup/go.md]: 👍

File: content/en/tracing/setup/go.md (hunk 2 above)
Comment (knusbaum) [reply content/en/tracing/setup/go.md]: 👍

File: content/en/tracing/setup/go.md (hunk 1 above)
Comment (knusbaum) [reply content/en/tracing/setup/go.md]: Huh, sorry about that. Not sure how I missed this. I `grep`'d to find this to begin with.

File: content/en/tracing/setup/go.md [hunk 4]
```diff
@@ -139,6 +139,31 @@ func main() {
 
 For more tracer settings, see available options in the [configuration documentation][57].
 
+### B3 Headers Extraction and Injection
+
+Datadog APM tracer supports [B3 headers extraction][58] and injection for distributed tracing.
+
+Distributed headers injection and extraction is controlled by
+configuring injection/extraction styles. Currently two styles are
+supported:
```
Comment (gbbr) [inline content/en/tracing/setup/go.md]: Should we just: 
```suggestion
supported: `Datadog` and `B3`.
```
And drop the list below? I'm not sure if it helps much.

Comment (knusbaum) [reply content/en/tracing/setup/go.md]: Yeah, I think that's fine.
A list makes sense if we ever add more, but we don't need to format the docs in anticipation of that.

File: content/en/tracing/setup/go.md [hunk 5]
```diff
@@ -139,6 +139,31 @@ func main() {
...
+`DD_PROPAGATION_STYLE_INJECT=Datadog,B3`
+
+Extraction styles can be configured using the environment variable
+`DD_PROPAGATION_STYLE_EXTRACT=Datadog,B3`
+
+The values of these environment variables are comma separated lists of
+header styles that are enabled for injection or extraction. By default only
+Datadog extraction style is enabled.
+
+If multiple extraction styles are enabled extraction attempt is done
```
[the remark this answers is not in the document]
Comment (knusbaum) [reply content/en/tracing/setup/go.md]: 👍

File: content/en/tracing/setup/go.md [hunk 6]
```diff
@@ -139,6 +139,31 @@ func main() {
...
+on the order those styles are configured and first successful
```
[the remark this answers is not in the document]
Comment (knusbaum) [reply content/en/tracing/setup/go.md]: 👍

File: content/en/tracing/setup/go.md [hunk 7]
```diff
@@ -139,6 +139,28 @@ func main() {
 
 For more tracer settings, see available options in the [configuration documentation][57].
 
+### B3 Headers Extraction and Injection
+
+Datadog APM tracer supports [B3 headers extraction][58] and injection for distributed tracing.
```
Comment (gbbr) [inline content/en/tracing/setup/go.md]:
```suggestion
The Datadog APM tracer supports [B3 headers extraction][58] and injection for distributed tracing.
```

File: content/en/tracing/setup/go.md [hunk 8]
```diff
@@ -139,6 +139,28 @@ func main() {
...
+
+Injection styles can be configured using the environment variable
+`DD_PROPAGATION_STYLE_INJECT=Datadog,B3`
+
+Extraction styles can be configured using the environment variable
+`DD_PROPAGATION_STYLE_EXTRACT=Datadog,B3`
+
+The values of these environment variables are comma separated lists of
+header styles that are enabled for injection or extraction. By default only
+Datadog extraction style is enabled.
```
Comment (gbbr) [inline content/en/tracing/setup/go.md]:
```suggestion
the `Datadog` extraction style is enabled.
```

File: content/en/tracing/setup/go.md [hunk 9]
```diff
@@ -139,6 +139,28 @@ func main() {
...
+
+If multiple extraction styles are enabled, extraction attempts are made
+in the order that those styles are specified and the first successful
```
Comment (gbbr) [inline content/en/tracing/setup/go.md]:
```suggestion
in the order that those styles are specified and the first successfully
```

File: content/en/tracing/setup/go.md [hunk 10]
```diff
@@ -139,6 +139,28 @@ func main() {
 
 For more tracer settings, see available options in the [configuration documentation][57].
 
+### B3 Headers Extraction and Injection
+
+The Datadog APM tracer supports [B3 headers extraction][58] and injection for distributed tracing.
+
+Distributed headers injection and extraction is controlled by
+configuring injection/extraction styles. Currently two styles are
```
Comment (cswatt) [inline content/en/tracing/setup/go.md]:
```suggestion
configuring injection/extraction styles. Two styles are
```

File: content/en/tracing/setup/go.md [hunk 11]
```diff
@@ -139,6 +139,28 @@ func main() {
...
+supported: `Datadog` and `B3`.
+
+Injection styles can be configured using the environment variable
```
Comment (cswatt) [inline content/en/tracing/setup/go.md]:
```suggestion
Configure injection styles using the environment variable
```

File: content/en/tracing/setup/go.md [hunk 12]
```diff
@@ -139,6 +139,28 @@ func main() {
...
+`DD_PROPAGATION_STYLE_INJECT=Datadog,B3`
+
+Extraction styles can be configured using the environment variable
```
Comment (cswatt) [inline content/en/tracing/setup/go.md]:
```suggestion
Configure extraction styles using the environment variable
```

File: content/en/tracing/setup/go.md [hunk 13]
```diff
@@ -139,6 +139,28 @@ func main() {
...
+`DD_PROPAGATION_STYLE_EXTRACT=Datadog,B3`
+
+The values of these environment variables are comma separated lists of
+header styles that are enabled for injection or extraction. By default only
+the `Datadog` extraction style is enabled.
+
+If multiple extraction styles are enabled, extraction attempts are made
+in the order that those styles are specified and the first successfully
```
Comment (cswatt) [inline content/en/tracing/setup/go.md]:
```suggestion
in the order that those styles are specified. The first successfully
```

**Pull request merged**